mne/preprocessing/stim.py

```diff
@@ -67,3 +101,81 @@ def eliminate_stim_artifact(raw, events, event_id, tmin=-0.005,
         elif mode == 'window':
             raw[picks, first_samp:last_samp] = data * window[np.newaxis, :]
     return raw
+
+
+def fix_stim_artifact(inst, events=None, event_id=None, tmin=0.,
+                      tmax=0.01, mode='linear', copy=False):
+    """Eliminate stimulation's artifacts from instance
+
+    Parameters
+    ----------
+    inst : instance of Raw or Evoked or Epochs
+        instance
+    events : array, shape (n_events, 3)
+        The list of events. Required only when inst is raw
+    event_id : int
+        The id of the events generating the stimulation artifacts.
+        Required only when inst is raw
+    tmin : float
+        Start time of the interpolation window in seconds.
+    tmax : float
+        End time of the interpolation window in seconds.
+    mode : 'linear' | 'window'
+        way to fill the artifacted time interval.
+        'linear' does linear interpolation
+        'window' applies a (1 - hanning) window.
+    copy : bool
+        If False inst is cropped in place.
+
+    Returns
+    -------
+    inst : instance of Raw or Evoked or Epochs
+        instance with modified data
+    """
+    if copy:
+        inst = inst.copy()
+    s_start = int(np.ceil(inst.info['sfreq'] * tmin))
+    s_end = int(np.ceil(inst.info['sfreq'] * tmax))
+    if (s_end - s_start) < 4:
+        mode = 'linear'  # When time is too short, force linear mode.
+    window = None
+    if mode == 'window':
+        window = _get_window(s_start, s_end)
+    ch_names = inst.info['ch_names']
+    picks = pick_channels(ch_names, ch_names)
+
+    if isinstance(inst, Raw):
+        _check_preload(inst)
+        events_sel = (events[:, 2] == event_id)
+        event_start = events[events_sel, 0]
+        data, _ = inst[:, :]
+        for event_idx in event_start:
+            first_samp = int(event_idx) - inst.first_samp + s_start
+            last_samp = int(event_idx) - inst.first_samp + s_end
+            _fix_artifact(inst, data, window, picks, first_samp,
+                          last_samp, mode)
+
+    elif isinstance(inst, Epochs):
+        _check_preload(inst)
+        if inst.reject:
+            raise RuntimeError('Reject is already applied. Use reject=None '
+                               'in the constructor.')
+        e_start = int(np.ceil(inst.info['sfreq'] * inst.tmin))
+        first_samp = s_start - e_start
+        last_samp = s_end - e_start
+        data = inst.get_data()[:, picks, :]
+        for epoch, k in zip(data, range(np.shape(data)[2])):
+            _fix_artifact(inst.get_data()[k], epoch, window, picks, first_samp,
+                          last_samp, mode)
```

Comment:
you see that if you fix the ._data attribute it will work all the time as long as preload=True.

Again I would just do this.